packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,452 @@
 
 part of firebase_ml_vision;
 
+/// Detector for performing optical character recognition(OCR) on an input image.
+///
+/// A barcode detector is created via getVisionBarcodeDetector() in [FirebaseVision]:
+///
+/// ```dart
+/// BarcodeDetector barcodeDetector = FirebaseVision.instance.barcodeDetector();
+/// ```
 class BarcodeDetector extends FirebaseVisionDetector {
-  BarcodeDetector._(BarcodeDetectorOptions options);
+  BarcodeDetector._();
 
+  /// Closes the barcode detector and release its model resources.
   @override
   Future<void> close() async {
-    // TODO: implement close
+    return FirebaseVision.channel.invokeMethod('BarcodeDetector#close');
   }
 
+  /// Detects barcode in the input image.
+  ///
+  /// The OCR is performed asynchronously.
   @override
   Future<void> detectInImage(FirebaseVisionImage visionImage) async {
-    // TODO: implement detectInImage
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<VisionBarcode> barcodes = <VisionBarcode>[];
+    reply.forEach((dynamic barcode) {
+      barcodes.add(new VisionBarcode._(barcode));
+    });
+
+    return barcodes;
   }
 }
 
-class BarcodeDetectorOptions {}
+class VisionBarcode {
```

Comment:
Could you add [dartdoc](https://www.dartlang.org/guides/language/effective-dart/documentation) documentation explaining what the class is for? Also could you add documentation for all the public variables?

Typically, you can just copy and paste the documentation used on the firebase website for [Barcodes](https://firebase.google.com/docs/reference/android/com/google/firebase/ml/vision/barcode/FirebaseVisionBarcode).  

You can look at TextDetector as well for documentation examples.

Reply:
I got it!